src/hotspot/share/prims/whitebox.cpp

```diff
@@ -2131,6 +2131,15 @@ WB_ENTRY(jint, WB_GetCDSCurrentVersion(JNIEnv* env, jobject wb))
 #endif
 WB_END
 
+WB_ENTRY(jint, WB_GetArchiveRelocationMode(JNIEnv* env, jobject wb))
+#if INCLUDE_CDS
+  return (jint)ArchiveRelocationMode;
+#else
+  ShouldNotReachHere();
+  return (jint)-1;
```

Comment:
Unnecessary dead code.  Or maybe this shouldn't be using `ShouldNotReachHere()`. Is the intent
to crash in a debug build but return an error code in a product build?  If so, `ShouldNotReachHere()`
doesn't provide that behavior, as it affects product builds too.

Reply:
I think the intent is to crash if someone runs a test that uses CDS and this API in a build without CDS available. You literally should not reach here as the test should have been skipped at a higher level that is true for debug and product test runs.